__tests__/app.test.js

```diff
@@ -231,3 +297,67 @@ describe('/api/users', () => {
         })
     });
 });
+
+describe('/api/articles queries', () => {
+    test('GET request responds with 200 and all articles of one topic', () => {
+        return request(app)
+        .get('/api/articles?topic=cats')
+        .expect(200)
+        .then(( { body }) => {
+            body.articles.forEach(article=> {
+                expect(article).toMatchObject({
+                    article_id: expect.any(Number),
+                    topic: 'cats',
+                    author: expect.any(String),
+                    created_at: expect.any(String),
+                    votes: expect.any(Number)
+                })
+            })
+        }) 
+    });
+
+    test('GET request responds with 200 and all articles ordered by asc or desc', () => {
+        return request(app)
+        .get('/api/articles?order=asc')
+        .expect(200)
+        .then(( { body }) => {
+            body.articles.forEach(article=> {
+                expect(article).toMatchObject({
+                    article_id: expect.any(Number),
+                    topic: expect.any(String),
+                    author: expect.any(String),
+                    created_at: expect.any(String),
+                    votes: expect.any(Number)
+                })
+            })
+            expect(body.articles).toBeSorted('created_at',{ascending: true})
+        }) 
+    });
+
+    test('GET request responds with 200 and all articles sorted by a specific column', () => {
+        return request(app)
+        .get('/api/articles?sort_by=votes')
+        .expect(200)
+        .then(( { body }) => {
+            body.articles.forEach(article=> {
+                expect(article).toMatchObject({
+                    article_id: expect.any(Number),
+                    topic: expect.any(String),
+                    author: expect.any(String),
+                    created_at: expect.any(String),
+                    votes: expect.any(Number)
+                })
+            })
+            expect(body.articles).toBeSortedBy('votes', {descending: true})
+        }) 
+    });
+
+    test('GET request responds with 404 if sort_by column is not valid', () => {
+        return request(app)
+        .get('/api/articles?sort_by=moo')
+        .expect(404)
+        .then(( { body }) => {
```

Comment:
This is definitely one of those borderline ones... but I would probably say this is erring on the side of a 400. I would argue it is a bad request, because it is contradicting the schema (i.e. the structure of the tables - there is no column with a name like this). I would generally save a 404 for when it can't find a specific _row_ in the database, i.e. there is no actual data corresponding, rather than it being in contradiction with the actual database setup as a whole